validation/src/main/java/io/serverlessworkflow/validation/WorkflowValidatorImpl.java

```diff
@@ -61,298 +62,273 @@ public WorkflowValidator setSource(String source) {
   }
 
   @Override
-  public List<ValidationError> validate() {
+  public Collection<ValidationError> validate() {
     validationErrors.clear();
-    if (workflow == null) {
+    if (workflow == null && schemaValidationEnabled && source != null) {
       try {
-        if (schemaValidationEnabled && source != null) {
-          try {
-            if (!source.trim().startsWith("{")) {
-              // convert yaml to json to validate
-              ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
-              Object obj = yamlReader.readValue(source, Object.class);
-
-              ObjectMapper jsonWriter = new ObjectMapper();
-
-              workflowSchema.validate(new JSONObject(jsonWriter.writeValueAsString(obj)));
-            } else {
-              workflowSchema.validate(new JSONObject(source));
-            }
-          } catch (ValidationException e) {
-            e.getCausingExceptions().stream()
-                .map(ValidationException::getMessage)
-                .forEach(
-                    m -> {
-                      if ((!m.equals("#/functions: expected type: JSONObject, found: JSONArray")
-                          && !m.equals("#/events: expected type: JSONObject, found: JSONArray")
-                          && !m.equals("#/start: expected type: JSONObject, found: String")
-                          && !m.equals("#/retries: expected type: JSONObject, found: JSONArray"))) {
-                        addValidationError(m, ValidationError.SCHEMA_VALIDATION);
-                      }
-                    });
-          }
-        }
-      } catch (Exception e) {
-        logger.error("Schema validation exception: " + e.getMessage());
+
+        JsonSchemaFactory.getInstance(VersionFlag.V202012)
```

Comment:
Isn't [V7](https://github.com/networknt/json-schema-validator/blob/master/src/main/java/com/networknt/schema/SpecVersion.java#L26) the most used? Maybe we should use the same one defined in the old dependency?

Reply:
V7 is fine, I believe old one was V2, but was unsure about it